runtime/command-start/src/main/java/io/aklivity/zilla/runtime/command/start/internal/airline/ZillaStartCommand.java

```diff
@@ -109,6 +109,22 @@ public void run()
         }
 
         EngineConfiguration config = new EngineConfiguration(props);
+        if ("file".equals(config.configURL().getProtocol()))
+        {
+            if (config.configURL().toString().endsWith("zilla.yaml") &&
+                    Files.notExists(Paths.get(config.configURL().getPath())))
+            {
+                String jsonConfig = Paths.get(config.configURL().getPath()).getParent() != null ?
+                        Paths.get(config.configURL().getPath()).getParent() + "/zilla.json" : "zilla.json";
+                props.setProperty(ENGINE_CONFIG_URL.name(), "file:" + jsonConfig);
+                config = new EngineConfiguration(props);
+                System.out.println("zilla.yaml file not found, loading zilla.json instead");
+            }
+            if (config.configURL().toString().endsWith(".json"))
+            {
+                System.out.println("warning: json syntax is deprecated, migrate to yaml");
+            }
+        }
```

Comment:
This block of code calls `config.configURL()` 6 times, seems like we should hoist it into a variable, no? 😄 

First, rename field named `configURL` of type `URI` to `configURI` instead.

```
EngineConfiguration config = new EngineConfiguration(props);
URL configURL = config.configURL();
if ("file".equals(configURL.getProtocol()))
{
    Path configPath = Paths.get(configURL.getPath());
    if (configPath.endsWith("zilla.yaml") && Files.notExists(configPath))
    {
        String configJSON = String.format("file:%s", configPath.resolveSibling("zilla.json"));
        props.setProperty(ENGINE_CONFIG_URL.name(), configJSON);
        config = new EngineConfiguration(props);
        System.out.println("zilla.yaml file not found, loading zilla.json instead");
    }
    if (configPath.getFilename().getName(0).endsWith(".json"))
    {
        System.out.println("warning: json syntax is deprecated, migrate to yaml");
    }
}
```

Reply:
Also, is reinstantiation of `config = new EngineConfiguration(props);` actually necessary, or is it sufficient to just update properties?